### PR TITLE
logger: add logger_status orb message

### PR DESCRIPTION
**Pull request by dagar**
Resurrecting this old branch I never finished. The idea is to expose logger status over orb both for the simplicity of monitoring over a regular flight and for deeper integration with the state machine (optional arming requirements, https://github.com/PX4/Firmware/pull/13200).

WORK IN PROGRESS

### CODE REVIEW

**Comment by dagar** [comment]
I had some other questions and ideas here about things like broadcasting status (eg is logger ready to start before arming) and capturing dropout duration differently for the log rather than the way it's reset with logger status now, but I'd rather get some basic in now we can continue to improve incrementally.

**Comment by dagar** [comment]
> Adding a status is ok, but it won't help with #13200,

I know, I addressed that above. Next I'll extend this to publish status pre-logging to handle things like #13200 (it's actually a fairly common request).

I also think it's going to be worth it to casually monitor logging rate and buffer usage over time across a large range of configurations and different phases of flight. Yes we could get it from the log itself, but we largely don't.

Somewhat related, I've seen a number of logs that look like data was missed, but don't have dropouts. I'm wondering about tracking generation gaps in full rate topics (no interval configured) and exposing it here. The overall priorities need a review.

File: src/modules/logger/logger.cpp [hunk 1]
```diff
@@ -785,11 +785,34 @@ void Logger::run()
 				_last_sync_time = loop_time;
 			}
 
-			// update buffer statistics
-			for (int i = 0; i < (int)LogType::Count; ++i) {
-				if (!_statistics[i].dropout_start && (_writer.get_buffer_fill_count_file((LogType)i) > _statistics[i].high_water)) {
-					_statistics[i].high_water = _writer.get_buffer_fill_count_file((LogType)i);
```
Comment (bkueng) [inline src/modules/logger/logger.cpp]: Why did you remove this? This is now broken.

Please don't just merge PR's w/o review.

Comment (dagar) [reply src/modules/logger/logger.cpp]: Ouch, yes it's clearly bogus, I have it fixed in the wip followup, but I can bring it to master sooner if needed. I apparently screwed up the rebase.

I would much prefer everything goes into master with a real review, but at the moment quite a lot goes ignored. I'd be up for establishing a more thorough system, but we'd need to do something like figure out code owners (and backups), commit to reasonable response times, etc. I also don't want to encourage more "rubber stamp" code reviews where we have the illusion of good process with the overhead and bureaucracy, but hardly any of the actual benefit.
